src/js/integrations/reactnativeerrorhandlers.ts

```diff
@@ -58,15 +58,28 @@ export class ReactNativeErrorHandlers implements Integration {
     if (this._options.onunhandledrejection) {
       /*
         In newer RN versions >=0.63, the global promise is not the same reference as the one imported from the promise library.
-        This is due to a version mismatch between promise versions. The version will need to be fixed with a package resolution.
-        We first run a check and show a warning if needed.
+        This is due to a version mismatch between promise versions.
       */
-      this._checkPromiseVersion();
+      this._polyfillPromise();
+    }
+  }
+  /**
+   * Polyfill the global promise instance with one we can be sure that we can attach the tracking to.
+   */
+  private _polyfillPromise(): void {
+    /* eslint-disable import/no-extraneous-dependencies,@typescript-eslint/no-var-requires*/
+    const {
+      polyfillGlobal,
+    } = require("react-native/Libraries/Utilities/PolyfillFunctions");
+    const Promise = require("promise/setimmediate/es6-extensions");
```

Comment:
Do we have to try catch this? What about checking for `global.Promise`?

Reply:
try catching require statements don't work in older versions of React Native and the hack I tried using last time had undefined behavior so I want to avoid doing so now. These dependencies should be present since our minimum required Rn version 0.56.